CesiumAsync/include/CesiumAsync/Impl/ContinuationFutureType.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "CesiumAsync/Impl/ContinuationReturnType.h"
+#include "CesiumAsync/Impl/RemoveFuture.h"
+
+namespace CesiumAsync {
+
+template <typename T> class Future;
+
+namespace Impl {
+// Begin omitting doxgen warnings for Impl namespace
+//! @cond Doxygen_Suppress
+
+template <typename Func, typename T> struct ContinuationFutureType {
+  typedef Future<typename RemoveFuture<
+      typename ContinuationReturnType<Func, T>::type>::type>
+      type;
```

Comment:
slightly nickpick from me, but `using` keyword is more readable in this case, and since we use it a little below, it will be consistent